engine/crates/engine-v2/src/engine.rs

```diff
@@ -29,22 +35,97 @@ impl Engine {
     }
 
     pub async fn execute(&self, request: engine::Request, headers: RequestHeaders) -> Response {
-        let operation = match self.prepare(&request) {
+        let coordinator = match self.prepare(request, headers) {
+            Ok(ok) => ok,
+            Err(response) => return response,
+        };
+
+        if matches!(coordinator.operation_type(), OperationType::Subscription) {
+            return Response::from_error(
+                GraphqlError::new("Subscriptions are only suported on streaming transports.  Try making a request with SSE or WebSockets"),
+                ExecutionMetadata::default(),
+            );
+        }
+
+        coordinator.execute().await
+    }
+
+    pub fn execute_stream(
+        &self,
+        request: engine::Request,
+        headers: RequestHeaders,
+    ) -> impl Stream<Item = Response> + '_ {
```

Comment:
Considering how the cli federation dev is structured you'll most likely need to make it `'static` with an `Arc<Engine>` I think

Reply:
Yeah, you may be right - I'll deal with it when I get to the inevitable compiler errors.